packages/terra-hookshot/src/Hookshot.jsx

```diff
@@ -0,0 +1,336 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Portal from 'react-portal';
+import HookshotContent from './HookshotContent';
+import HookshotUtils from './_HookshotUtils';
+
+const ATTACHMENT_POSITIONS = [
+  'top left',
+  'top center',
+  'top right',
+  'middle left',
+  'middle center',
+  'middle right',
+  'bottom left',
+  'bottom center',
+  'bottom right',
+];
+
+const ATTACHMENT_BEHAVIORS = [
+  'auto',
+  'flip',
+  'none',
+];
+
+const propTypes = {
+  /**
+   * How the content should be positioned when the primary attachment is not available.
+   * Valid values: 'auto', 'flip', and 'none'.
+   */
+  attachmentBehavior: PropTypes.oneOf(ATTACHMENT_BEHAVIORS),
+  /**
+   * Value in px of the margin to place between the target and the content.
+   */
+  attachmentMargin: PropTypes.number,
+  /**
+   * Reference to the bounding container, will use window if nothing is provided.
+   */
+  boundingRef: PropTypes.func,
+  /**
+   * The HookshotContent to be attached.
+   */
+  content: PropTypes.element.isRequired,
+  /**
+   * String pair separated by a space using values of top, middle, bottom, and left, center, right. (ex: 'top middle')
+   */
+  contentAttachment: PropTypes.oneOf(ATTACHMENT_POSITIONS).isRequired,
+  /**
+   * String pair of top and left offset, ie "10px -4px".
+   */
+  contentOffset: PropTypes.string,
```

Comment:
contentOffset to me sounds like it should only expect 1 offset being given to it; however, here you are specifying that two should be given in the string.  Seems like this should be broken out into two separate offsets or change the name to better resemble that its two offets

Reply:
Still debating this one, are other input?

Reply:
Could it be an object itself?  contentOffset.top and contentOffset.left?

Reply:
What happens if someone provides a string with only one value like `-10px`? Does the left offset default to `0` or would one need to pass in `-10px 0`?

Two props may provide a clean api for defaults, the single prop taking an object could also provide a default object. I lean towards the two props as I think its more friendly to the use case of providing a single direction offset without hindering multi-direction offset.